File: Week6/Valid-Palindrome/hwq.java
public class Solution {
    public boolean isPalindrome(String s) {
        s=s.replaceAll("[^0-9a-zA-Z]","");
        s=s.toLowerCase();
        StringBuffer sb=new StringBuffer(s);
        String s2=sb.reverse().toString();
        return s.equals(s2);
    }
}

File: Week6/Valid-Palindrome/hwq.py
class Solution:
    # @param s, a string
    # @return a boolean
    def isPalindrome(self, s):
        p = re.compile('[^0-9a-zA-Z]')
        s=p.sub('',s)
        s=s.lower()             
        c=s[::-1]
        return c==s
